mithril-test-lab/mithril-end-to-end/src/main.rs

```diff
@@ -198,9 +208,66 @@ async fn main() -> StdResult<()> {
 
     app_stopper.stop().await;
     join_set.shutdown().await;
+
     res
 }
 
+#[derive(Debug)]
+enum AppResult {
+    Success(),
+    UnretryableError(anyhow::Error),
+    RetryableError(anyhow::Error),
```

Comment:
```suggestion
    RetryableError(anyhow::Error),
    Cancelled { reason: String },
```
I think that a `Cancelled` variant is missing to properly handle user cancellation (`ctrl+c` etc..., it should return a `ExitCode::FAILURE` result code and be used in `with_gracefull_shutdown` below).

Reply:
I added the support of cancellation with signals for `AppResult` in https://github.com/input-output-hk/mithril/pull/2145/commits/6fd68f146c3f825e93770f6650915ed867a8cf36 :+1: